src/lib.rs

```diff
@@ -72,13 +72,31 @@ macro_rules! kernel_module {
         )*
     };
 
-    (@attribute $name:ident, $value:expr) => {
+    (@attribute author, $value:expr) => {
         #[link_section = ".modinfo"]
-        #[allow(non_upper_case_globals)]
-        // TODO: Generate a name the same way the kernel's `__MODULE_INFO` does.
-        // TODO: This needs to be a `[u8; _]`, since the kernel defines this as a  `const char []`.
-        // See https://github.com/rust-lang/rfcs/pull/2545
-        pub static $name: &'static [u8] = concat!(stringify!($name), "=", $value, '\0').as_bytes();
+        pub static AUTHOR_KEY: [u8; 7] = *b"author=";
```

Comment:
In the kernel's implementation, these are concat'd together and assigned to a single value:

https://github.com/torvalds/linux/blob/219d54332a09e8d8741c1e1982f5eae56099de85/include/linux/moduleparam.h#L23-L26

DOes doing 3 separate statics achieve hte same impact?

Reply:
It definitely works in practice (try running `modinfo`).

It works because they all end up in the same section, right after each other, and the NUL terminator is only on the last one. It's a kind of silly approach, and I don't know if there's a _guarantee_ that rustc or ld will lay them out in order, but it's a trick I've seen before in C.

Reply:
Can you add a comment explaining why/how it works?

Do you think a test that shelled out to modinfo would be practical?